src/main/java/wolox/training/models/Book.java

```diff
@@ -86,70 +87,72 @@ public String getAuthor() {
     }
 
     public void setAuthor(String author) {
-        this.author = author;
+        this.author = Preconditions.checkNotNull(author, "The author can't be null");
     }
 
     public String getImage() {
         return image;
     }
 
     public void setImage(String image) {
-        this.image = image;
+        this.image = Preconditions.checkNotNull(image, "The image can't be null");
     }
 
     public String getTitle() {
         return title;
     }
 
     public void setTitle(String title) {
-        this.title = title;
+        this.title = Preconditions.checkNotNull(title, "The title can't be null");
     }
 
     public String getSubtitle() {
         return subtitle;
     }
 
     public void setSubtitle(String subtitle) {
-        this.subtitle = subtitle;
+        this.subtitle = Preconditions.checkNotNull(subtitle, "The subtitle can't be null");
     }
 
     public String getPublisher() {
         return publisher;
     }
 
     public void setPublisher(String publisher) {
-        this.publisher = publisher;
+        this.publisher = Preconditions.checkNotNull(publisher, "The publisher can't be null");
     }
 
     public String getYear() {
         return year;
     }
 
     public void setYear(String year) {
-        this.year = year;
+        this.year = Preconditions.checkNotNull(year, "The year can't be null");
     }
 
     public int getPages() {
         return pages;
     }
 
     public void setPages(int pages) {
-        this.pages = pages;
+        this.pages = Preconditions.checkNotNull(pages, "The pages can't be null");
     }
 
     public String getIsbn() {
         return isbn;
     }
 
     public void setIsbn(String isbn) {
-        this.isbn = isbn;
+        this.isbn = Preconditions.checkNotNull(isbn, "The isbn can't be null");
     }
 
     public List<User> getUsers() {
         return (List<User>) Collections.unmodifiableCollection(users);
     }
 
     public void setUsers(List<User> users) {
+        Preconditions.checkArgument(users != null && !users.isEmpty(),
```

Comment:
This could be empty, no worries with that. A book doesn't depend on a user to exist.